src/address.rs

```diff
@@ -176,3 +176,90 @@ macro_rules! impl_address_ops {
         }
     };
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd)]
+    pub struct MockAddress(pub u64);
+    impl_address_ops!(MockAddress, u64);
+
+    #[test]
+    fn test_offset_from() {
+        let base = MockAddress(0x100);
+        let addr = MockAddress(0x150);
+        assert_eq!(addr.unchecked_offset_from(base), 0x50u64);
+        assert_eq!(addr.checked_offset_from(base), Some(0x50u64));
+        assert_eq!(base.checked_offset_from(addr), None);
+    }
+
+    #[test]
+    fn test_equals() {
+        let a = MockAddress(0x300);
+        let b = MockAddress(0x300);
+        let c = MockAddress(0x301);
+        assert_eq!(a, MockAddress(a.raw_value()));
+        assert_eq!(a, b);
+        assert_eq!(b, a);
+        assert_ne!(a, c);
+        assert_ne!(c, a);
+    }
+
+    #[test]
+    #[allow(clippy::eq_op)]
+    fn test_cmp() {
+        let a = MockAddress(0x300);
+        let b = MockAddress(0x301);
+        assert!(a < b);
```

Comment:
This is a bit weird and clippy also complains about it. Let's just check that `a < b` instead.

Reply:
Fixed.